src/components/Providers/useAccountForProvider.spec.jsx

```diff
@@ -0,0 +1,179 @@
+import React from 'react'
+import { renderHook } from '@testing-library/react-hooks'
+
+import AppLike from 'test/AppLike'
+
+import useAccountForProvider from './useAccountForProvider'
+import { saveAccountToSettings } from './helpers'
+
+jest.mock('./helpers', () => ({
+  saveAccountToSettings: jest.fn()
+}))
+
+const accountsData = [{ _id: 'accountId01' }, { _id: 'accountId02' }]
+const settingsData = [{ account: { _id: 'accountId03' } }]
+
+const setup = ({ settings, accounts, selectedAccount } = {}) => {
+  const wrapper = ({ children }) => <AppLike>{children}</AppLike>
+
+  return renderHook(
+    () => useAccountForProvider(settings, accounts, selectedAccount),
+    {
+      wrapper
+    }
+  )
+}
+
+describe('useAccountForProvider', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('should return null if settings and accounts are loading', async () => {
```

Comment:
C'est `renderHook` qui requière que les tests soient asynchrones ?

Reply:
saveAccountToSettings is async, I guess

Reply:
but using async without await is useless

Reply:
I think you should add https://eslint.org/docs/rules/require-await in the project

Reply:
non c'est un copier/coller raté (qui a embarqué le async)